Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3361:23: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3361 |   static gpgme_ctx_t *new_gpgme_ctx_t_p() {
#      |                       ^~~~~~~~~~~~~~~~~
# 3359|   
# 3360|   
# 3361|->   static gpgme_ctx_t *new_gpgme_ctx_t_p() { 
# 3362|       return (gpgme_ctx_t *)calloc(1,sizeof(gpgme_ctx_t));
# 3363|     }

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3382:24: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3382 |   static gpgme_data_t *new_gpgme_data_t_p() {
#      |                        ^~~~~~~~~~~~~~~~~~
# 3380|   
# 3381|   
# 3382|->   static gpgme_data_t *new_gpgme_data_t_p() { 
# 3383|       return (gpgme_data_t *)calloc(1,sizeof(gpgme_data_t));
# 3384|     }

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3403:23: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3403 |   static gpgme_key_t *new_gpgme_key_t_p() {
#      |                       ^~~~~~~~~~~~~~~~~
# 3401|   
# 3402|   
# 3403|->   static gpgme_key_t *new_gpgme_key_t_p() { 
# 3404|       return (gpgme_key_t *)calloc(1,sizeof(gpgme_key_t));
# 3405|     }

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3424:25: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3424 |   static gpgme_error_t *new_gpgme_error_t_p() {
#      |                         ^~~~~~~~~~~~~~~~~~~
# 3422|   
# 3423|   
# 3424|->   static gpgme_error_t *new_gpgme_error_t_p() { 
# 3425|       return (gpgme_error_t *)calloc(1,sizeof(gpgme_error_t));
# 3426|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_error_t_p_assign’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3437:10: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
# 3437 |     *obj = value;
#      |     ~~~~~^~~~~~~
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:44688:17: note: ‘arg2’ was declared here
#44688 |   gpgme_error_t arg2 ;
#      |                 ^~~~
# 3435|   
# 3436|     static void gpgme_error_t_p_assign(gpgme_error_t *obj, gpgme_error_t value) {
# 3437|->     *obj = value;
# 3438|     }
# 3439|   

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3445:30: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3445 |   static gpgme_trust_item_t *new_gpgme_trust_item_t_p() {
#      |                              ^~~~~~~~~~~~~~~~~~~~~~~~
# 3443|   
# 3444|   
# 3445|->   static gpgme_trust_item_t *new_gpgme_trust_item_t_p() { 
# 3446|       return (gpgme_trust_item_t *)calloc(1,sizeof(gpgme_trust_item_t));
# 3447|     }

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3466:31: warning[-Wstrict-prototypes]: function declaration isn’t a prototype
# 3466 |   static gpgme_engine_info_t *new_gpgme_engine_info_t_p() {
#      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~
# 3464|   
# 3465|   
# 3466|->   static gpgme_engine_info_t *new_gpgme_engine_info_t_p() { 
# 3467|       return (gpgme_engine_info_t *)calloc(1,sizeof(gpgme_engine_info_t));
# 3468|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_data_seek’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:3633:14: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
# 3633 |     result = gpgme_data_seek(arg1,arg2,arg3);
#      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 3631|     {
# 3632|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 3633|->     result = gpgme_data_seek(arg1,arg2,arg3);
# 3634|       SWIG_PYTHON_THREAD_END_ALLOW;
# 3635|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_strerror’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:4176:22: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
# 4176 |     result = (char *)gpgme_strerror(arg1);
#      |                      ^~~~~~~~~~~~~~~~~~~~
# 4174|     {
# 4175|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 4176|->     result = (char *)gpgme_strerror(arg1);
# 4177|       SWIG_PYTHON_THREAD_END_ALLOW;
# 4178|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_strerror_r’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:4230:14: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
# 4230 |     result = (int)gpgme_strerror_r(arg1,arg2,arg3);
#      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 4228|     {
# 4229|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 4230|->     result = (int)gpgme_strerror_r(arg1,arg2,arg3);
# 4231|       SWIG_PYTHON_THREAD_END_ALLOW;
# 4232|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_strsource’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:4270:22: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
# 4270 |     result = (char *)gpgme_strsource(arg1);
#      |                      ^~~~~~~~~~~~~~~~~~~~~
# 4268|     {
# 4269|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 4270|->     result = (char *)gpgme_strsource(arg1);
# 4271|       SWIG_PYTHON_THREAD_END_ALLOW;
# 4272|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_err_code_to_errno’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:4329:14: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
# 4329 |     result = (int)gpgme_err_code_to_errno(arg1);
#      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 4327|     {
# 4328|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 4329|->     result = (int)gpgme_err_code_to_errno(arg1);
# 4330|       SWIG_PYTHON_THREAD_END_ALLOW;
# 4331|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_err_make_from_errno’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:4411:14: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
# 4411 |     result = gpgme_err_make_from_errno(arg1,arg2);
#      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# 4409|     {
# 4410|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 4411|->     result = gpgme_err_make_from_errno(arg1,arg2);
# 4412|       SWIG_PYTHON_THREAD_END_ALLOW;
# 4413|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_key_sig_status_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:7667:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
# 7667 |     if (arg1) (arg1)->status = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
# 7665|     {
# 7666|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
# 7667|->     if (arg1) (arg1)->status = arg2;
# 7668|       SWIG_PYTHON_THREAD_END_ALLOW;
# 7669|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_invalid_key_reason_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:11077:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#11077 |     if (arg1) (arg1)->reason = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
#11075|     {
#11076|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#11077|->     if (arg1) (arg1)->reason = arg2;
#11078|       SWIG_PYTHON_THREAD_END_ALLOW;
#11079|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_io_event_done_data_err_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:12710:27: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#12710 |     if (arg1) (arg1)->err = arg2;
#      |               ~~~~~~~~~~~~^~~~~~
#12708|     {
#12709|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#12710|->     if (arg1) (arg1)->err = arg2;
#12711|       SWIG_PYTHON_THREAD_END_ALLOW;
#12712|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_io_event_done_data_op_err_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:12775:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#12775 |     if (arg1) (arg1)->op_err = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
#12773|     {
#12774|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#12775|->     if (arg1) (arg1)->op_err = arg2;
#12776|       SWIG_PYTHON_THREAD_END_ALLOW;
#12777|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_data_new_from_filepart’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:14711:14: warning[-Wmaybe-uninitialized]: ‘arg4’ may be used uninitialized in this function
#14711 |     result = gpgme_data_new_from_filepart(arg1,(char const *)arg2,arg3,arg4,arg5);
#      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#14709|     {
#14710|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#14711|->     result = gpgme_data_new_from_filepart(arg1,(char const *)arg2,arg3,arg4,arg5);
#14712|       SWIG_PYTHON_THREAD_END_ALLOW;
#14713|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_recipient_status_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:19611:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#19611 |     if (arg1) (arg1)->status = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
#19609|     {
#19610|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#19611|->     if (arg1) (arg1)->status = arg2;
#19612|       SWIG_PYTHON_THREAD_END_ALLOW;
#19613|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_signature_status_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:25176:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#25176 |     if (arg1) (arg1)->status = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
#25174|     {
#25175|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#25176|->     if (arg1) (arg1)->status = arg2;
#25177|       SWIG_PYTHON_THREAD_END_ALLOW;
#25178|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_signature_validity_reason_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:25793:39: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#25793 |     if (arg1) (arg1)->validity_reason = arg2;
#      |               ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
#25791|     {
#25792|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#25793|->     if (arg1) (arg1)->validity_reason = arg2;
#25794|       SWIG_PYTHON_THREAD_END_ALLOW;
#25795|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_import_status_result_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:27966:30: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#27966 |     if (arg1) (arg1)->result = arg2;
#      |               ~~~~~~~~~~~~~~~^~~~~~
#27964|     {
#27965|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#27966|->     if (arg1) (arg1)->result = arg2;
#27967|       SWIG_PYTHON_THREAD_END_ALLOW;
#27968|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_op_export_ext_start’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:30597:19: warning[-Wmaybe-uninitialized]: ‘size2’ may be used uninitialized in this function
#30597 |     for (i = 0; i < size2; i++)
#      |                 ~~^~~~~~~
#30595|       size_t i;
#30596|       free(vector2);
#30597|->     for (i = 0; i < size2; i++)
#30598|       Py_XDECREF(pyVector2[i]);
#30599|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_op_export_ext’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:30912:19: warning[-Wmaybe-uninitialized]: ‘size2’ may be used uninitialized in this function
#30912 |     for (i = 0; i < size2; i++)
#      |                 ~~^~~~~~~
#30910|       size_t i;
#30911|       free(vector2);
#30912|->     for (i = 0; i < size2; i++)
#30913|       Py_XDECREF(pyVector2[i]);
#30914|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_op_keylist_ext_start’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:36067:19: warning[-Wmaybe-uninitialized]: ‘size2’ may be used uninitialized in this function
#36067 |     for (i = 0; i < size2; i++)
#      |                 ~~^~~~~~~
#36065|       size_t i;
#36066|       free(vector2);
#36067|->     for (i = 0; i < size2; i++)
#36068|       Py_XDECREF(pyVector2[i]);
#36069|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_op_spawn_start’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:37601:19: warning[-Wmaybe-uninitialized]: ‘size3’ may be used uninitialized in this function
#37601 |     for (i = 0; i < size3; i++)
#      |                 ~~^~~~~~~
#37599|       size_t i;
#37600|       free(vector3);
#37601|->     for (i = 0; i < size3; i++)
#37602|       Py_XDECREF(pyVector3[i]);
#37603|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_gpgme_op_spawn’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:37745:19: warning[-Wmaybe-uninitialized]: ‘size3’ may be used uninitialized in this function
#37745 |     for (i = 0; i < size3; i++)
#      |                 ~~^~~~~~~
#37743|       size_t i;
#37744|       free(vector3);
#37745|->     for (i = 0; i < size3; i++)
#37746|       Py_XDECREF(pyVector3[i]);
#37747|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap__gpgme_op_assuan_result_err_set’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:43998:27: warning[-Wmaybe-uninitialized]: ‘arg2’ may be used uninitialized in this function
#43998 |     if (arg1) (arg1)->err = arg2;
#      |               ~~~~~~~~~~~~^~~~~~
#43996|     {
#43997|       SWIG_PYTHON_THREAD_BEGIN_ALLOW;
#43998|->     if (arg1) (arg1)->err = arg2;
#43999|       SWIG_PYTHON_THREAD_END_ALLOW;
#44000|     }

Error: COMPILER_WARNING (CWE-457):
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c: scope_hint: In function ‘_wrap_copy_gpgme_error_t_p’
gpgme-1.15.1/lang/python/python3.10-gpg/gpgme_wrap.c:44631:17: warning[-Wmaybe-uninitialized]: ‘arg1’ may be used uninitialized in this function
#44631 |   gpgme_error_t arg1 ;
#      |                 ^~~~
#44629|   SWIGINTERN PyObject *_wrap_copy_gpgme_error_t_p(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
#44630|     PyObject *resultobj = 0;
#44631|->   gpgme_error_t arg1 ;
#44632|     PyObject *swig_obj[1] ;
#44633|     gpgme_error_t *result = 0 ;

Error: COMPILER_WARNING:
gpgme-1.15.1/lang/qt/tests/t-verify.cpp: scope_hint: In member function 'void VerifyTest::testSignatureKey()'
gpgme-1.15.1/lang/qt/tests/t-verify.cpp:84:25: warning[-Wrange-loop-construct]: loop variable 'subkey' creates a copy from type 'const GpgME::Subkey'
#   84 |         for (const auto subkey: key.subkeys()) {
#      |                         ^~~~~~
gpgme-1.15.1/lang/qt/tests/t-verify.cpp:84:25: note: use reference type to prevent copying
#   84 |         for (const auto subkey: key.subkeys()) {
#      |                         ^~~~~~
#      |                         &
#   82|   
#   83|           bool found = false;
#   84|->         for (const auto subkey: key.subkeys()) {
#   85|               if (!strcmp (subkey.fingerprint(), sig.fingerprint())) {
#   86|                   found = true;

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c: scope_hint: In function 'add_data'
gpgme-1.15.1/src/engine-gpg.c:313:4: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  313 |   a->next = NULL;
#      |    ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  311|     if (!a)
#  312|       return gpg_error_from_syserror ();
#  313|->   a->next = NULL;
#  314|     a->data = data;
#  315|     a->inbound = inbound;

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:314:4: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  314 |   a->data = data;
#      |    ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  312|       return gpg_error_from_syserror ();
#  313|     a->next = NULL;
#  314|->   a->data = data;
#  315|     a->inbound = inbound;
#  316|     a->arg_locp = NULL;

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:315:4: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  315 |   a->inbound = inbound;
#      |    ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  313|     a->next = NULL;
#  314|     a->data = data;
#  315|->   a->inbound = inbound;
#  316|     a->arg_locp = NULL;
#  317|   

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:316:4: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  316 |   a->arg_locp = NULL;
#      |    ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  314|     a->data = data;
#  315|     a->inbound = inbound;
#  316|->   a->arg_locp = NULL;
#  317|   
#  318|     if (dup_to == -2)

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:320:8: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  320 |       a->print_fd = 1;
#      |        ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  318|     if (dup_to == -2)
#  319|       {
#  320|->       a->print_fd = 1;
#  321|         a->dup_to = -1;
#  322|       }

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:321:8: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  321 |       a->dup_to = -1;
#      |        ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  319|       {
#  320|         a->print_fd = 1;
#  321|->       a->dup_to = -1;
#  322|       }
#  323|     else

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:325:8: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  325 |       a->print_fd = 0;
#      |        ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  323|     else
#  324|       {
#  325|->       a->print_fd = 0;
#  326|         a->dup_to = dup_to;
#  327|       }

Error: COMPILER_WARNING (CWE-788):
gpgme-1.15.1/src/engine-gpg.c:326:8: warning[-Warray-bounds]: array subscript 'struct arg_and_data_s[0]' is partly outside array bounds of 'unsigned char[47]'
#  326 |       a->dup_to = dup_to;
#      |        ^~
gpgme-1.15.1/src/engine-gpg.c:310:7: note: referencing an object of size 47 allocated by 'malloc'
#  310 |   a = malloc (sizeof *a - 1);
#      |       ^~~~~~~~~~~~~~~~~~~~~~
#  324|       {
#  325|         a->print_fd = 0;
#  326|->       a->dup_to = dup_to;
#  327|       }
#  328|     *gpg->argtail = a;
